src/components/organisms/AppRouter/AdminSubrouter.tsx

```diff
@@ -5,31 +5,100 @@ import Admin from "pages/Admin/Admin";
 import { AdminAdvancedSettings } from "pages/AdminAdvancedSettings";
 
 import { AdminVenueView } from "components/organisms/AdminVenueView";
+import { Provided } from "components/organisms/AppRouter/Provided";
 
 export const AdminSubrouter: React.FC = () => {
   return (
     <Switch>
       {/* Admin V1 */}
-      <Route path="/admin/venue/rooms/:venueId" component={RoomsForm} />
-      <Route path="/admin/venue/creation" component={VenueWizard} />
-      <Route path="/admin/venue/edit/:venueId" component={VenueWizard} />
-      <Route path="/admin/:venueId" component={Admin} />
-      <Route path="/admin" component={Admin} />
+      <Route
+        path="/admin/venue/rooms/:venueId"
+        component={() => (
+          <Provided withWorldUsers withRelatedVenues>
+            <RoomsForm />
+          </Provided>
+        )}
+      />
+      <Route
+        path="/admin/venue/creation"
+        component={
+          // for some state related reasons, Provided fails this VenueWizard
+          // and using function here will cause the error
+          // so WithNavigationBar wraps the RelatedVenuesProvider for it
+          VenueWizard
+        }
+      />
+      <Route
+        path="/admin/venue/edit/:venueId"
+        component={() => (
+          <Provided withWorldUsers withRelatedVenues>
+            <VenueWizard />
+          </Provided>
+        )}
+      />
+      <Route
+        path="/admin/:venueId"
+        component={() => (
+          <Provided withWorldUsers withRelatedVenues>
+            <Admin />
+          </Provided>
+        )}
+      />
+      <Route
+        path="/admin"
+        component={() => (
+          <Provided withWorldUsers withRelatedVenues>
+            <Admin />
+          </Provided>
+        )}
+      />
 
       {/* Admin V2/3/NG */}
-      <Route path="/admin-ng/venue/:venueId?" component={AdminVenueView} />
+      <Route
+        path="/admin-ng/venue/:venueId?"
+        component={() => (
+          <Provided withWorldUsers>
+            <AdminVenueView />
+          </Provided>
+        )}
+      />
       <Route
         path="/admin-ng/advanced-settings/:venueId?"
-        component={AdminAdvancedSettings}
+        component={() => (
+          <Provided withWorldUsers withRelatedVenues>
+            <AdminAdvancedSettings />
+          </Provided>
+        )}
+      />
+      <Route
+        path="/admin-ng/venue/creation"
+        component={() => (
+          <Provided withWorldUsers withRelatedVenues>
+            <VenueWizardV2 />
+          </Provided>
+        )}
+      />
+      <Route
+        path="/admin-ng/edit/:venueId"
+        component={() => (
+          <Provided withWorldUsers withRelatedVenues>
+            <VenueWizardV2 />
+          </Provided>
+        )}
+      />
+      <Route
+        path="/admin-ng"
+        component={() => (
+          <Provided withWorldUsers>
+            <AdminV2 />
+          </Provided>
+        )}
```

Comment:
This `component={() => (` syntax is invalid and will cause constant re-renders.

The 'nested' method I used in my described solution is the canonically best/correct way to do this.

**Further Reading:**

- https://reactrouter.com/web/api/Route/route-render-methods
  - > **The recommended method of rendering something with a `<Route>` is to use children elements**, as shown above. There are, however, a few **other methods** you can use to render something with a `<Route>`. These are provided mostly for **supporting apps that were built with earlier versions of the router before hooks were introduced.**
  - https://reactrouter.com/web/api/Route/component
    - > When you use component (instead of render or children, below) the router uses `React.createElement` to create a new React element from the given component. That means **if you provide an inline function to the component prop, you would create a new component every render. This results in the existing component unmounting and the new component mounting instead of just updating the existing component.** When using an inline function for inline rendering, use the `render` or the `children` prop (below).
- https://reactrouter.com/web/api/Route/render-func
  - > This allows for convenient inline rendering and wrapping **without the undesired remounting explained above.**

Reply:
> The 'nested' method I used in my described solution is the canonically best/correct way to do this.

Had spent more than an hour trying to figure out why it's either _TypeScript_ complaining for the type or the router in the browser throwing _functions aren't components_ errors.

I did try `render` instead of `component` prop in the `Route`, that should reduce re-renders, but went with `component` instead due to lack of time to properly test the differences.

But I'm open to solution that would create something akin to the original suggestion i.e. some `<ProvidedRoute>` that will closely mimic the `<Route>` interface

Reply:
> Started with the "canonical solution". Had spent more than an hour trying to figure out why it's either TypeScript complaining for the type or the router in the browser throwing functions aren't components errors.

Are you able to reproduce that now? And if so, can you please show me the exact code that would cause that?

Reply:
> I'm open to solution that would create something akin to the original suggestion i.e. some `<ProvidedRoute>` that will closely mimic the `<Route>` interface

This was my secondary solution that I decided was a worse idea than my primary solution.

Reply:
fixed in https://github.com/sparkletown/sparkle/pull/1769/commits/c58428242104ab1c0becf206460260f3c346a754

This doesn't seem to throw errors, and is the exact solution I described originally, so not sure what you were doing to get the errors?